src/js/stores/ChronosStore.js

```diff
@@ -0,0 +1,117 @@
+import {EventEmitter} from 'events';
+
+import AppDispatcher from '../events/AppDispatcher';
+import ChronosActions  from '../events/ChronosActions';
+import {
+  CHRONOS_JOBS_CHANGE,
+  CHRONOS_JOBS_ERROR,
+  VISIBILITY_CHANGE
+} from '../constants/EventTypes';
+import Config from '../config/Config';
+import JobTree from '../structs/JobTree';
+import {
+  REQUEST_CHRONOS_JOBS_ERROR,
+  REQUEST_CHRONOS_JOBS_ONGOING,
+  REQUEST_CHRONOS_JOBS_SUCCESS,
+  SERVER_ACTION
+} from '../constants/ActionTypes';
+
+import VisibilityStore from './VisibilityStore';
+
+let requestInterval;
+
+function startPolling() {
+  if (!requestInterval) {
+    ChronosActions.fetchJobs();
+    requestInterval = global.setInterval(
+      ChronosActions.fetchJobs,
+      Config.getRefreshRate()
+    );
+  }
+}
+
+function stopPolling() {
+  if (requestInterval) {
+    global.clearInterval(requestInterval);
+    requestInterval = null;
+  }
+}
+
+function handleInactiveChange() {
+  let isInactive = VisibilityStore.get('isInactive');
+  if (isInactive) {
+    stopPolling();
+  }
+
+  if (!isInactive && ChronosStore.shouldPoll()) {
```

Comment:
Apologies for the comment on a merged PR, but I don't think there is an instantiated ChronosStore in scope here, so I think this will cause reference errors.